### PR TITLE
new catalog item: cowcheck

### PR DESCRIPTION
**Pull request by wjimenez5271**

### CODE REVIEW

File: templates/cowcheck/0/rancher-compose.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+version: '2'
+catalog:
+  name: cowcheck
+  version: 0.2.0
+  description: A microservice for checking the health of a Rancher node
+  minimum_rancher_version: v.1.6.0
+  questions:
```
Comment (rawmind0) [inline templates/cowcheck/0/rancher-compose.yml]: Questions are bad indented. Could you please indent one to the left??

File: templates/cowcheck/0/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      DOCKER_API_VERSION: ${DOCKER_API_VERSION}
+      DATA_SPACE_THRESHOLD: ${DATA_SPACE_THRESHOLD}
+      METADATA_SPACE_THRESHOLD: ${METADATA_SPACE_THRESHOLD}
+      POLL_INTERVAL: ${POLL_INTERVAL}
+    stdin_open: true
+    tty: true
+    ports:
+    - 5050:5050/tcp
+    labels:
+      io.rancher.container.pull_image: always
```
Comment (rawmind0) [inline templates/cowcheck/0/docker-compose.yml]: You don't need to pull image always.
```
-      io.rancher.container.pull_image: always
```

File: templates/cowcheck/0/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    environment:
+      LOG_LEVEL: ${LOG_LEVEL}
+      DOCKER_API_VERSION: ${DOCKER_API_VERSION}
+      DATA_SPACE_THRESHOLD: ${DATA_SPACE_THRESHOLD}
+      METADATA_SPACE_THRESHOLD: ${METADATA_SPACE_THRESHOLD}
+      POLL_INTERVAL: ${POLL_INTERVAL}
+    stdin_open: true
+    tty: true
+    ports:
+    - 5050:5050/tcp
```
Comment (rawmind0) [inline templates/cowcheck/0/docker-compose.yml]: What do you think to add a question to add a question for published port??
```
-    - 5050:5050/tcp
+    - ${PUBLISH_PORT}:5050/tcp
```

File: templates/cowcheck/0/rancher-compose.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      - variable: METADATA_SPACE_THRESHOLD
+        label: METADATA_SPACE_THRESHOLD
+        description: | Minimum amount of free Docker Data Storage space in bytes
+        type: string
+        default: 1.24
+      - variable: DOCKER_API_VERSION
+        label: DOCKER_API_VERSION
+        description: | Version of the Docker API to use (storage checks only)
+        type: string
+        default: 1.24
```
Comment (rawmind0) [inline templates/cowcheck/0/rancher-compose.yml]: Publish port question
```
+    - variable: PUBLISH_PORT
+      label: PUBLISH_PORT
+      description: | Port to publish the service.
+      type: int
+      default: 5050
```

**Comment by rawmind0** [comment]
Hi @wjimenez5271 , thanks so much for the PR. :)

Build is failing due to bad questions indent. Could you please take a look at changes requested??

**Comment by wjimenez5271** [comment]
@rawmind0 I've updated the PR, however it still failing but I can't tell why

File: templates/cowcheck/0/rancher-compose.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+version: '2'
+catalog:
+  name: cowcheck
+  version: 0.2.0
+  description: A microservice for checking the health of a Rancher node
+  minimum_rancher_version: v.1.6.0
+  questions:
+    - variable: POLL_INTERVAL
+      label: POLL_INTERVAL
+      description: | Frequency in seconds to run checks
```
Comment (rawmind0) [inline templates/cowcheck/0/rancher-compose.yml]: Description fields are making build fails. If you use `|` you need new line.
Please, review all description fields.
```
-      description: | Frequency in seconds to run checks
+      description: | 
+        Frequency in seconds to run checks
OR
+      description: Frequency in seconds to run checks
```

File: templates/cowcheck/0/rancher-compose.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      type: string
+      default: 1.24
+    - variable: ENABLE_STORAGE_CHECK
+      label: ENABLE_STORAGE_CHECK
+      description: | Enable storage check
+      type: bool
```
Comment (rawmind0) [inline templates/cowcheck/0/rancher-compose.yml]: wrong type `bool`, `boolean` instead.
```
-      type: bool
+      type: boolean
```

**Comment by rawmind0** [comment]
@wjimenez5271, build is failing now due to description fields at rancher-compose. Please, take a look at the changes requested. 

Also, could you please add `templates/cowcheck/catalogIcon-cowcheck.svg` (prefered) or `templates/cowcheck/catalogIcon-cowcheck.png` icon to show into the catalog entry??

File: templates/cowcheck/0/rancher-compose.yml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  name: cowcheck
+  version: 0.2.0
+  description: A microservice for checking the health of a Rancher node
+  minimum_rancher_version: v.1.6.0
+  questions:
+    - variable: POLL_INTERVAL
+      label: POLL_INTERVAL
+      description: Frequency in seconds to run checks
+      type: string
+      default: 1.24
```
Comment (rawmind0) [inline templates/cowcheck/0/rancher-compose.yml]: This value makes deployment fail, may be better a default value of 5, 10, 30....??
```
-      default: 1.24
+      default: 10
```

File: templates/cowcheck/0/rancher-compose.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    - variable: ENABLE_STORAGE_CHECK
+      label: ENABLE_STORAGE_CHECK
+      description: Enable storage check
+      type: boolean
+      default: false
+    - variable: DATA_SPACE_THRESHOLD
+      label: DATA_SPACE_THRESHOLD
+      description: Minimum amount of free Docker Data Storage space in bytes
+      type: string
+      default: 1.24
```
Comment (rawmind0) [inline templates/cowcheck/0/rancher-compose.yml]: If this value is the minimum amount of free bytes, may be better a default value of 1024, 10240...??
```
-      default: 1.24
+      default: 1024
```

File: templates/cowcheck/0/rancher-compose.yml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    - variable: METADATA_SPACE_THRESHOLD
+      label: METADATA_SPACE_THRESHOLD
+      description: Minimum amount of free Docker Data Storage space in bytes
+      type: string
+      default: 1.24
```
Comment (rawmind0) [inline templates/cowcheck/0/rancher-compose.yml]: If this value is the minimum amount of free bytes, may be better a default value of 1024, 10240...??
```
-      default: 1.24
+      default: 1024
```

File: templates/cowcheck/0/rancher-compose.yml [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+      type: boolean
+      default: false
+    - variable: DATA_SPACE_THRESHOLD
+      label: DATA_SPACE_THRESHOLD
+      description: Minimum amount of free Docker Data Storage space in bytes
+      type: string
+      default: 1.24
+    - variable: METADATA_SPACE_THRESHOLD
+      label: METADATA_SPACE_THRESHOLD
+      description: Minimum amount of free Docker Data Storage space in bytes
```
Comment (rawmind0) [inline templates/cowcheck/0/rancher-compose.yml]: Typo error
```
-      description: Minimum amount of free Docker Data Storage space in bytes
+      description: Minimum amount of free Docker Metadata Storage space in bytes
```

File: templates/cowcheck/0/rancher-compose.yml [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+      type: string
+      default: 1.24
+    - variable: DOCKER_API_VERSION
+      label: DOCKER_API_VERSION
+      description: Version of the Docker API to use (storage checks only)
+      type: string
+      default: 1.24
```
Comment (rawmind0) [inline templates/cowcheck/0/rancher-compose.yml]: If this value is the docker api version, may be better a default value of 1.12, 1.13,...??
```
-      default: 1.24
+      default: 1.12
```

**Comment by rawmind0** [comment]
Thanks for the update @wjimenez5271...

If i deploy the packages with default values, i got these error logs...
```
21/12/2017 10:35:34panic: non-positive interval for NewTicker
21/12/2017 10:35:34
21/12/2017 10:35:34goroutine 20 [running]:
21/12/2017 10:35:34time.NewTicker(0x0, 0x3)
21/12/2017 10:35:34	/usr/local/Cellar/go/1.9/libexec/src/time/tick.go:23 +0x1d0
21/12/2017 10:35:34main.checkPoller(0xc42030c2a0, 0x3, 0x3, 0x0)
21/12/2017 10:35:34	/Users/wjimenez/go/src/github.com/wjimenez5271/cowcheck/main.go:270 +0x58
21/12/2017 10:35:34created by main.main
21/12/2017 10:35:34	/Users/wjimenez/go/src/github.com/wjimenez5271/cowcheck/main.go:333 +0x33d
```

If i change default values, the package run correctly. Could you please take a look to the changes requested to set correct default values??

**Comment by wjimenez5271** [comment]
@rawmind0 sorry Raul, I don't know how I missed those defaults, you're right they are completely wrong

File: templates/cowcheck/0/rancher-compose.yml (hunk 11 above)
Comment (wjimenez5271) [reply templates/cowcheck/0/rancher-compose.yml]: I've only tested on 1.24 so for now can we keep it this way?

Comment (rawmind0) [reply templates/cowcheck/0/rancher-compose.yml]: Sure, if default value works, it's ok. I think i misunderstand docker version with docker API version.

**Comment by rawmind0** [comment]
No problem @wjimenez5271 ... Right now is ok.. :)

LGTM...

**Pull request merged**